karmadactl: add the ability to automatically generate user guidance documents for the command line (#2260)
karmada-io/karmada: docs/userguide/commands/karmadactl_promote.md

## karmadactl promote

Promote resources from legacy clusters to karmada control plane

### Synopsis

Promote resources from legacy clusters to karmada control plane. Requires the cluster be joined or registered.

```
karmadactl promote <RESOURCE_TYPE> <RESOURCE_NAME> -n <NAME_SPACE> -C <CLUSTER_NAME> [flags]
```

### Examples

```
  # Promote deployment(default/nginx) from cluster1 to Karmada
  karmadactl promote deployment nginx -n default -C cluster1
  
  # Promote deployment(default/nginx) with gvk from cluster1 to Karmada
  karmadactl promote deployment.v1.apps nginx -n default -C cluster1
  

>   # Dumps the artifacts but does not deploy them to Karmada, same as 'dry run'

Reviewer: ```suggestion
  # Dump the artifacts but not deploy them to Karmada, same as 'dry run'
```

  karmadactl promote deployment nginx -n default -C cluster1 -o yaml|json
  
  # Promote secret(default/default-token) from cluster1 to Karmada
  karmadactl promote secret default-token -n default -C cluster1
  
  # Support to use '--cluster-kubeconfig' to specify the configuration of member cluster
  karmadactl promote deployment nginx -n default -C cluster1 --cluster-kubeconfig=<CLUSTER_KUBECONFIG_PATH>
  
  # Support to use '--cluster-kubeconfig' and '--cluster-context' to specify the configuration of member cluster
  karmadactl promote deployment nginx -n default -C cluster1 --cluster-kubeconfig=<CLUSTER_KUBECONFIG_PATH> --cluster-context=<CLUSTER_CONTEXT>
```

### Options

```

>   -C, --cluster string              the name of legacy cluster (eg -C=member1)

Reviewer: ```suggestion
  -C, --cluster string              Name of legacy cluster (e.g. -C=member1)
```

      --cluster-context string      Context name of legacy cluster in kubeconfig. Only works when there are multiple contexts in the kubeconfig.
      --cluster-kubeconfig string   Path of the legacy cluster's kubeconfig.
      --dry-run                     Run the command in dry-run mode, without making any server requests.

>   -h, --help                        help for promote

Reviewer: ```suggestion
  -h, --help                        Help for promote
```

      --karmada-context string      Name of the cluster context in control plane kubeconfig file.
  -n, --namespace string            -n=namespace or -n namespace (default "default")

>   -o, --output string               Output format. One of: json|yaml

Reviewer: ```suggestion
  -o, --output string               Output format, json or yaml
```

```

### Options inherited from parent commands

```

>       --add-dir-header                   If true, adds the file directory to the header of the log messages

Reviewer: pls make the same changes to this part as above.
